lib/matplotlib/units.py

```diff
@@ -114,61 +115,28 @@ def is_numlike(x):
 
 
 class Registry(dict):
-    """
-    register types with conversion interface
-    """
-    def __init__(self):
-        dict.__init__(self)
-        self._cached = {}
+    """Register types with conversion interface."""
 
     def get_converter(self, x):
-        'get the converter interface instance for x, or None'
-
-        if not len(self):
```

Comment:
I think it would be good to keep this in, to avoid stepping through the following logic if there aren't any convertors registered in the first place.

Reply:
I think the cost of an instance check, a dict lookup and attempting to get the first element is quite small.  For code like this I would like to see at least a microbenchmark showing at least some minor improvement (not necessarily a big one) before making it more complex than necessary.